tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+
+For this step you'll understand the following concepts of OAM:
+
+- `Component`: describes an individual container or other workload
+- `ApplicationConfiguration`: describes the deployment of a group of components
+
+---
+
+Here's the basic `Component` for this step - the contents are printed here with comments for explanation purposes - you can find this example saved at `./hello-world/component.arm`. For now focus on reading and understanding these examples, we'll deploy them in a minute.
+
+The container used in this example is a "Hello, world!" type webserver sample from nginx.
+
+```js
+// Resource statement declare an object - every type of OAM definition in ARM-DSL will be inside a resource statement.
+// The string 'Component@core.oam.dev/v1alpha2' specifies the kind of resource and its version.
+resource oam 'Component@core.oam.dev/v1alpha2' _ {
+
+  // The metadata section must specify a name for the object. 
+  metadata: {
+    name: 'hello-world'
+  }
+  
+  // The spec section describes the properties of the component.
+  spec : {
+
+    // The workload section allows the description of different types of workloads.
+    //
+    // Right now the only supported workload is 'ContainerizedWorkload' - which is intended
+    // for long-running server type containers.
+    //
+    // Workloads each have kind and apiVersion as well as a name. For now just remember that the
+    // name should be the the same as the component name.
+    workload: {
+      apiVersion: 'core.oam.dev/v1alpha2'
+      kind: 'ContainerizedWorkload'
+      metadata: {
+        name: 'hello-world'
+      }
+
+      // The spec section (inside the workload) contains the properties that configure the container.
+      spec: {
+
+        // Our workloads are all running on linux-amd64 but other platforms can be supported.
+        osType: 'linux'
+        arch: 'amd64'
+
+        // A list of containers to run and their relevant settings. This should be familiar
+        // if you've ever used docker-compose or Kubernetes.
+        containers: [
+          {
+            name: 'hello-world'
+            image: 'nginxdemos/hello'
+            ports: [
+              {
+                name: 'http'
+                containerPort: 80
+                protocol: 'TCP'
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+ARM-DSL is a JSON-like language intended for tersely specifying objects. Arrays are defined with `[]`, objects are defined with `{}`, strings are *single-quoted* (`'this is a string'`).
+
+Defining a `Component` is like specifying a *template for a container* - nothing runs as a result of defining a `Component`. What's needed to deploy the application is to define an `ApplicationConfiguration`. 
+
+The `ApplicationConfiguration` used in this step has its contents displayed here for learning purposes. You can find it on disk at `./hello-world/applicationconfiguration.arm`.
```

Comment:
We have not asked the customer to checkout the code at any point, so there is nothing on disk. Ideally, we should avoid requiring code checkout to play the tutorial. The URL to the file should be enough here.

Reply:
I am going to add this to the instructions. There's ARM-DSL code in the repo we expect the user to use, and we might need to scale this up to reduce the length of the tutorial or make other tweaks. It's not a lot to ask and probable is less clumsy than downloading files manually.

Reply:
The URL makes it easy to see on the browser. This makes the instructions more readable.